app/javascript/controllers/textarea_autogrow_controller.js

```diff
@@ -0,0 +1,39 @@
+import { Controller } from '@hotwired/stimulus'
+
+export default class extends Controller {
+
+  initialize () {
+    this.onResize = undefined
+    this.resizeDebounceDelayValue = 100
+    this.autogrow = this.autogrow.bind(this)
+  }
+
+  connect () {
+    this.element.style.overflow = 'hidden'
+    this.onResize = this.resizeDebounceDelayValue > 0 ? debounce(this.autogrow, this.resizeDebounceDelayValue) : this.autogrow
+
+    this.autogrow()
+
+    this.element.addEventListener('input', this.autogrow)
+    window.addEventListener('resize', this.onResize)
```

Comment:
I think you need to add `.bind(this)` to these event listeners, or `this` will be undefined when the function is called.

Reply:
Yes, I am still a little confused by the javascript this scoping. But on this particular example it seems to be working without an error. What makes you think it needs the bind(this)?

Reply:
@robacarp Yes, I am still a little confused by the javascript this scoping. But on this particular example it seems to be working without an error. What makes you think it needs the bind(this)?

Reply:
I re-wrote all these just to be safe. Explained more in a different comment.